File: exercises/ba1b.go
package exercises

import (
	"log"
	"strconv"

	"github.com/vgalaktionov/bioinformatics/lib"
)

func (ex Exercise) BA1B(params []string) []string {
	text := []byte(params[0])
	k, err := strconv.Atoi(params[1])
	if err != nil {
		log.Fatalf("%s", err)
	}

	result := lib.FrequentWords(text, k, 0)

	return result
}

File: exercises/ba1b.py

def foobar(text: str, pattern: str):
    return 1


def ba1b(params: list[str]):
    text, pattern = params
    return foobar(text, pattern)


def test_ba1b():
    expected = 'CATG GCAT'
    result = ba1b(['ACGTTGCATGTCGCATGATGCATGAGAGCT', '4'])
    assert expected == result
